### DISCUSSION TITLE
Deploy/dev/Kubernetes/Cassandra failing

### DISCUSSION THREAD

**Discussion by vayu068** [Sunbird-Ed/Community#103] [Installation] [answered]
Hi,

Using the release tag : release-4.6.0_RC4

Please find below complete Jenkins logs:

Checking out git https://github.com/project-sunbird/sunbird-devops.git into /var/lib/jenkins/workspace/Deploy/dev/Kubernetes/Cassandra@script/2b17fab10110704d031125b5ff6396246bdbc852b6799e400e2777264b895f3b to read pipelines/deploy/cassandra-deploy/Jenkinsfile
The recommended git tool is: NONE
No credentials specified
Cloning the remote Git repository
Using shallow clone with depth 1
Cloning repository https://github.com/project-sunbird/sunbird-devops.git
 > git init /var/lib/jenkins/workspace/Deploy/dev/Kubernetes/Cassandra@script/2b17fab10110704d031125b5ff6396246bdbc852b6799e400e2777264b895f3b # timeout=10
Fetching upstream changes from https://github.com/project-sunbird/sunbird-devops.git
 > git --version # timeout=10
 > git --version # 'git version 2.34.1'
 > git fetch --tags --force --progress --depth=1 -- https://github.com/project-sunbird/sunbird-devops.git +refs/heads/*:refs/remotes/origin/* # timeout=10
 > git config remote.origin.url https://github.com/project-sunbird/sunbird-devops.git # timeout=10
 > git config --add remote.origin.fetch +refs/heads/*:refs/remotes/origin/* # timeout=10
Avoid second fetch
 > git rev-parse origin/release-4.6.0_RC4^{commit} # timeout=10
 > git rev-parse release-4.6.0_RC4^{commit} # timeout=10
Checking out Revision 298c0e0cb57f5d8fa32bbc3676ae57c6c2127a6f (release-4.6.0_RC4)
 > git config core.sparsecheckout # timeout=10
 > git checkout -f 298c0e0cb57f5d8fa32bbc3676ae57c6c2127a6f # timeout=10
Commit message: "Issue #SB-28497 feat:updated enrollment list size env (#3180)"
First time build. Skipping changelog.
Loading library deploy-conf@heads/shared-lib
Attempting to resolve heads/shared-lib from remote references...
 > git --version # timeout=10
 > git --version # 'git version 2.34.1'
 > git ls-remote -- https://github.com/project-sunbird/sunbird-devops.git # timeout=10
The recommended git tool is: NONE
No credentials specified
Cloning the remote Git repository
Using shallow clone with depth 1
Avoid fetching tags
Cloning repository https://github.com/project-sunbird/sunbird-devops.git
 > git init /var/lib/jenkins/workspace/Deploy/dev/Kubernetes/Cassandra@libs/deploy-conf # timeout=10
Fetching upstream changes from https://github.com/project-sunbird/sunbird-devops.git
 > git --version # timeout=10
 > git --version # 'git version 2.34.1'
 > git fetch --no-tags --force --progress --depth=1 -- https://github.com/project-sunbird/sunbird-devops.git +refs/heads/*:refs/remotes/origin/* # timeout=10
 > git config remote.origin.url https://github.com/project-sunbird/sunbird-devops.git # timeout=10
 > git config --add remote.origin.fetch +refs/heads/*:refs/remotes/origin/* # timeout=10
Avoid second fetch
Checking out Revision 778da7ce7f97efd003eed98071921a837d209343 (778da7ce7f97efd003eed98071921a837d209343)
 > git config core.sparsecheckout # timeout=10
 > git checkout -f 778da7ce7f97efd003eed98071921a837d209343 # timeout=10
Commit message: "fix: move variables to be accessible across both try blocks"
First time build. Skipping changelog.
[Pipeline] Start of Pipeline
[Pipeline] node
Running on Jenkins in /var/lib/jenkins/workspace/Deploy/dev/Kubernetes/Cassandra
[Pipeline] {
[Pipeline] stage
[Pipeline] { (checkout public repo)
[Pipeline] checkout
The recommended git tool is: NONE
No credentials specified
Cloning the remote Git repository
Using shallow clone with depth 1
Cloning repository https://github.com/project-sunbird/sunbird-devops.git
 > git init /var/lib/jenkins/workspace/Deploy/dev/Kubernetes/Cassandra # timeout=10
Fetching upstream changes from https://github.com/project-sunbird/sunbird-devops.git
 > git --version # timeout=10
 > git --version # 'git version 2.34.1'
 > git fetch --tags --force --progress --depth=1 -- https://github.com/project-sunbird/sunbird-devops.git +refs/heads/*:refs/remotes/origin/* # timeout=10
 > git config remote.origin.url https://github.com/project-sunbird/sunbird-devops.git # timeout=10
 > git config --add remote.origin.fetch +refs/heads/*:refs/remotes/origin/* # timeout=10
Avoid second fetch
 > git rev-parse origin/release-4.6.0_RC4^{commit} # timeout=10
 > git rev-parse release-4.6.0_RC4^{commit} # timeout=10
Checking out Revision 298c0e0cb57f5d8fa32bbc3676ae57c6c2127a6f (release-4.6.0_RC4)
 > git config core.sparsecheckout # timeout=10
 > git checkout -f 298c0e0cb57f5d8fa32bbc3676ae57c6c2127a6f # timeout=10
Commit message: "Issue #SB-28497 feat:updated enrollment list size env (#3180)"
[Pipeline] }
[Pipeline] // stage
[Pipeline] ansiColor
[Pipeline] {

[Pipeline] stage
[Pipeline] { (get artifact)
[Pipeline] ansiColor
[Pipeline] {

[Pipeline] sh
+ echo Deploy/dev/Kubernetes/Cassandra
[Pipeline] sh
+ echo Deploy/dev/Kubernetes/Cassandra
[Pipeline] sh
+ echo Deploy/dev/Kubernetes/Cassandra
[Pipeline] stage
[Pipeline] { (parameter checks)
[Pipeline] copyArtifacts
Copied 2 artifacts from "ArtifactUpload » dev » Core » Cassandra" build number 30
[Pipeline] sh
+ jq -r .artifact_name metadata.json
[Pipeline] sh
+ jq -r .node_name metadata.json
[Pipeline] echo
[1m[33martifact_version not specified, using the artifact_version specified in metadata.json. [0m
[Pipeline] sh
+ jq -r .artifact_version metadata.json
[Pipeline] }
[Pipeline] // stage
[Pipeline] }

[Pipeline] // ansiColor
[Pipeline] sh
+ pwd
[Pipeline] ansiColor
[Pipeline] {

[Pipeline] echo
[1m[33mOption chosen is JenkinsJob, using the artifacts copied[0m
[Pipeline] }

[Pipeline] // ansiColor
[Pipeline] }
[Pipeline] // stage
[Pipeline] stage
[Pipeline] { (deploy artifact)
[Pipeline] sh
+ unzip cassandra_artifacts.zip:release-4.6.0_RC1_23630dee_3
Archive:  cassandra_artifacts.zip:release-4.6.0_RC1_23630dee_3
  inflating: cassandra-migration-0.0.1-SNAPSHOT-jar-with-dependencies.jar  
[Pipeline] echo
{env=dev, module=Kubernetes, jobName=Cassandra, absolute_job_path=ArtifactUpload/dev/Core/Cassandra, agent=master, artifact_name=cassandra_artifacts.zip, artifact_version=release-4.6.0_RC1_23630dee_3, currentWs=/var/lib/jenkins/workspace/Deploy/dev/Kubernetes/Cassandra, artifact=cassandra_artifacts.zip:release-4.6.0_RC1_23630dee_3, ansiblePlaybook=/var/lib/jenkins/workspace/Deploy/dev/Kubernetes/Cassandra/ansible/cassandra-deploy.yml, ansibleExtraArgs=--vault-password-file /var/lib/jenkins/secrets/vault-pass -v}
[Pipeline] ansiColor
[Pipeline] {

[Pipeline] stage
[Pipeline] { (ansible-run)
[Pipeline] echo
{env=dev, module=Kubernetes, jobName=Cassandra, absolute_job_path=ArtifactUpload/dev/Core/Cassandra, agent=master, artifact_name=cassandra_artifacts.zip, artifact_version=release-4.6.0_RC1_23630dee_3, currentWs=/var/lib/jenkins/workspace/Deploy/dev/Kubernetes/Cassandra, artifact=cassandra_artifacts.zip:release-4.6.0_RC1_23630dee_3, ansiblePlaybook=/var/lib/jenkins/workspace/Deploy/dev/Kubernetes/Cassandra/ansible/cassandra-deploy.yml, ansibleExtraArgs=--vault-password-file /var/lib/jenkins/secrets/vault-pass -v}
[Pipeline] dir
Running in /var/lib/jenkins/workspace/Deploy/dev/Kubernetes/Cassandra/private
[Pipeline] {
[Pipeline] echo
[1m[33mInfo: Branch override is enabled[0m
[Pipeline] checkout
The recommended git tool is: NONE
using credential github-private-repo
Cloning the remote Git repository
Using shallow clone with depth 1
Avoid fetching tags
Cloning repository https://github.com/vayu068/Sunbird_private.git
 > git init /var/lib/jenkins/workspace/Deploy/dev/Kubernetes/Cassandra/private # timeout=10
Fetching upstream changes from https://github.com/vayu068/Sunbird_private.git
 > git --version # timeout=10
 > git --version # 'git version 2.34.1'
using GIT_ASKPASS to set credentials 
 > git fetch --no-tags --force --progress --depth=1 -- https://github.com/vayu068/Sunbird_private.git +refs/heads/*:refs/remotes/origin/* # timeout=10
 > git config remote.origin.url https://github.com/vayu068/Sunbird_private.git # timeout=10
 > git config --add remote.origin.fetch +refs/heads/*:refs/remotes/origin/* # timeout=10
Avoid second fetch
 > git rev-parse origin/master^{commit} # timeout=10
Checking out Revision 6d8113b604861ea485408dc7aeb06e9d063a4119 (origin/master)
 > git config core.sparsecheckout # timeout=10
 > git checkout -f 6d8113b604861ea485408dc7aeb06e9d063a4119 # timeout=10
Commit message: "Update secrets.yml"
First time build. Skipping changelog.
[Pipeline] }
[Pipeline] // dir
[Pipeline] sh
+ rsync -Lkr /var/lib/jenkins/workspace/Deploy/dev/Kubernetes/Cassandra/private/ansible/inventory/dev/Kubernetes/common.yml /var/lib/jenkins/workspace/Deploy/dev/Kubernetes/Cassandra/private/ansible/inventory/dev/Kubernetes/hosts /var/lib/jenkins/workspace/Deploy/dev/Kubernetes/Cassandra/private/ansible/inventory/dev/Kubernetes/secrets.yml /var/lib/jenkins/workspace/Deploy/dev/Kubernetes/Cassandra/ansible/inventory/env/
+ [ -f /var/lib/jenkins/workspace/Deploy/dev/Kubernetes/Cassandra/ansible/inventory/env/kubernetes.yaml ]
+ ansible-playbook -i /var/lib/jenkins/workspace/Deploy/dev/Kubernetes/Cassandra/ansible/inventory/env/hosts /var/lib/jenkins/workspace/Deploy/dev/Kubernetes/Cassandra/ansible/cassandra-deploy.yml --vault-password-file /var/lib/jenkins/secrets/vault-pass -v
[0;34mNo config file found; using defaults[0m
[1;35m[WARNING]: Invalid characters were found in group names but not replaced, use[0m
[1;35m-vvvv to see details[0m
[1;35m[0m
[1;35m[WARNING]:  * Failed to parse /var/lib/jenkins/workspace/Deploy/dev/Kubernetes/[0m
[1;35mCassandra/ansible/inventory/env/hosts with yaml plugin: Syntax Error while[0m
[1;35mloading YAML.   did not find expected <document start>  The error appears to be[0m
[1;35min '/var/lib/jenkins/workspace/Deploy/dev/Kubernetes/Cassandra/ansible/inventor[0m
[1;35my/env/hosts': line 3, column 1, but may be elsewhere in the file depending on[0m
[1;35mthe exact syntax problem.  The offending line appears to be:  # Replace the[0m
[1;35mvalue deployer to your server login user name, for example ubuntu[0m
[1;35mansible_ssh_user=jenkins ^ here[0m
[1;35m[0m
[1;35m[WARNING]:  * Failed to parse /var/lib/jenkins/workspace/Deploy/dev/Kubernetes/[0m
[1;35mCassandra/ansible/inventory/env/hosts with ini plugin: /var/lib/jenkins/workspa[0m
[1;35mce/Deploy/dev/Kubernetes/Cassandra/ansible/inventory/env/hosts:144: Section[0m
[1;35m[core:children] includes undefined group: postgresql-slave[0m
[1;35m[0m
[1;35m[WARNING]: Unable to parse /var/lib/jenkins/workspace/Deploy/dev/Kubernetes/Cas[0m
[1;35msandra/ansible/inventory/env/hosts as an inventory source[0m
[1;35m[0m
[1;35m[WARNING]: No inventory was parsed, only implicit localhost is available[0m
[1;35m[0m
[1;35m[WARNING]: provided hosts list is empty, only localhost is available. Note that[0m
[1;35mthe implicit localhost does not match 'all'[0m
[1;35m[0m

PLAY [cassandra] ***************************************************************
[1;35m[WARNING]: While constructing a mapping from /var/lib/jenkins/workspace/Deploy/[0m
[1;35mdev/Kubernetes/Cassandra/ansible/inventory/env/group_vars/all.yml, line 4,[0m
[1;35mcolumn 1, found a duplicate dict key (app_es_snapshot_host). Using last defined[0m
[1;35mvalue only.[0m
[1;35m[0m
[1;35m[WARNING]: While constructing a mapping from /var/lib/jenkins/workspace/Deploy/[0m
[1;35mdev/Kubernetes/Cassandra/ansible/inventory/env/group_vars/all.yml, line 4,[0m
[1;35mcolumn 1, found a duplicate dict key (sunbird_es_host). Using last defined[0m
[1;35mvalue only.[0m
[1;35m[0m
[1;35m[WARNING]: While constructing a mapping from /var/lib/jenkins/workspace/Deploy/[0m
[1;35mdev/Kubernetes/Cassandra/ansible/inventory/env/group_vars/all.yml, line 4,[0m
[1;35mcolumn 1, found a duplicate dict key (log_es_snapshot_host). Using last defined[0m
[1;35mvalue only.[0m
[1;35m[0m

TASK [Gathering Facts] *********************************************************
[0;32mok: [10.0.2.102][0m

TASK [cassandra-deploy : checking for last failed migration] *******************
[0;33mchanged: [10.0.2.102] => {[0m
[0;33m    "changed": true,[0m
[0;33m    "cmd": "cqlsh -e 'SELECT version from sunbird.cassandra_migration_version where success=false ALLOW FILTERING ;' | tail -n +4 | head -n -2 | sort | tail -n1 | xargs",[0m
[0;33m    "delta": "0:00:00.463900",[0m
[0;33m    "end": "2022-08-24 13:42:32.641917",[0m
[0;33m    "rc": 0,[0m
[0;33m    "start": "2022-08-24 13:42:32.178017"[0m
[0;33m}[0m

TASK [cassandra-deploy : Dropping failed migration] ****************************
[0;36mskipping: [10.0.2.102] => {[0m
[0;36m    "changed": false,[0m
[0;36m    "skip_reason": "Conditional result was False"[0m
[0;36m}[0m

TASK [cassandra-deploy : Copy the migration jar] *******************************
[0;33mchanged: [10.0.2.102] => {[0m
[0;33m    "changed": true,[0m
[0;33m    "checksum": "6449acb4705e55036e4a8811479529661f9a977a",[0m
[0;33m    "dest": "/home/jenkins/cassandra-migration-0.0.1-SNAPSHOT-jar-with-dependencies.jar",[0m
[0;33m    "gid": 0,[0m
[0;33m    "group": "root",[0m
[0;33m    "md5sum": "45150522e21aab8505f965e2020aeb7c",[0m
[0;33m    "mode": "0644",[0m
[0;33m    "owner": "root",[0m
[0;33m    "size": 5005707,[0m
[0;33m    "src": "/home/jenkins/.ansible/tmp/ansible-tmp-1661348552.745352-15764-176911061307436/source",[0m
[0;33m    "state": "file",[0m
[0;33m    "uid": 0[0m
[0;33m}[0m

TASK [cassandra-deploy : Run the application] **********************************
[0;33mchanged: [10.0.2.102] => {[0m
[0;33m    "changed": true,[0m
[0;33m    "cmd": "java -cp \"cassandra-migration-0.0.1-SNAPSHOT-jar-with-dependencies.jar\" com.contrastsecurity.cassandra.migration.utils.MigrationScriptEntryPoint",[0m
[0;33m    "delta": "0:01:08.507353",[0m
[0;33m    "end": "2022-08-24 13:43:42.505596",[0m
[0;33m    "rc": 0,[0m
[0;33m    "start": "2022-08-24 13:42:33.998243"[0m
[0;33m}[0m
[0;33m[0m
[0;33mSTDOUT:[0m
[0;33m[0m
[0;33mMigration started at ==1661348554071[0m
[0;33mCassandra Migration 0.0.1-SNAPSHOT by Contrast Security[0m
[0;33mConnected to cluster: Test Cluster[0m
[0;33mData center: datacenter1; Host: /10.0.2.102[0m
[0;33mCurrent version of keyspace sunbird: 1.50[0m
[0;33mMigrating keyspace sunbird to version 1.51 - cassandra[0m
[0;33mMigrating keyspace sunbird to version 1.52 - cassandra[0m
[0;33mMigrating keyspace sunbird to version 1.53 - cassandra[0m
[0;33mMigrating keyspace sunbird to version 1.54 - cassandra[0m
[0;33mMigrating keyspace sunbird to version 1.55 - cassandra[0m
[0;33mMigrating keyspace sunbird to version 1.56 - cassandra[0m
[0;33mMigrating keyspace sunbird to version 1.57 - cassandra[0m
[0;33mMigrating keyspace sunbird to version 1.58 - cassandra[0m
[0;33mMigrating keyspace sunbird to version 1.59 - cassandra[0m
[0;33mMigrating keyspace sunbird to version 1.60 - cassandra[0m
[0;33mMigrating keyspace sunbird to version 1.61 - cassandra[0m
[0;33mMigrating keyspace sunbird to version 1.62 - cassandra[0m
[0;33mMigrating keyspace sunbird to version 1.63 - cassandra[0m
[0;33mMigrating keyspace sunbird to version 1.64 - cassandra[0m
[0;33mMigrating keyspace sunbird to version 1.65 - cassandra[0m
[0;33mMigrating keyspace sunbird to version 1.66 - cassandra[0m
[0;33mMigrating keyspace sunbird to version 1.67 - cassandra[0m
[0;33mMigrating keyspace sunbird to version 1.68 - cassandra[0m
[0;33mMigrating keyspace sunbird to version 1.69 - cassandra[0m
[0;33mMigrating keyspace sunbird to version 1.70 - cassandra[0m
[0;33mMigrating keyspace sunbird to version 1.71 - cassandra[0m
[0;33mMigrating keyspace sunbird to version 1.72 - cassandra[0m
[0;33mMigrating keyspace sunbird to version 1.73 - cassandra[0m
[0;33mMigrating keyspace sunbird to version 1.74 - cassandra[0m
[0;33mMigrating keyspace sunbird to version 1.75 - cassandra[0m
[0;33mMigrating keyspace sunbird to version 1.76 - cassandra[0m
[0;33mMigrating keyspace sunbird to version 1.77 - cassandra[0m
[0;33mMigrating keyspace sunbird to version 1.78 - cassandra[0m
[0;33mMigrating keyspace sunbird to version 1.79 - cassandra[0m
[0;33mMigrating keyspace sunbird to version 1.80 - cassandra[0m
[0;33mMigrating keyspace sunbird to version 1.81 - cassandra[0m
[0;33mMigrating keyspace sunbird to version 1.82 - cassandra[0m
[0;33mMigrating keyspace sunbird to version 1.83 - cassandra[0m
[0;33mMigrating keyspace sunbird to version 1.84 - cassandra[0m
[0;33mMigrating keyspace sunbird to version 1.85 - cassandra[0m
[0;33mMigrating keyspace sunbird to version 1.86 - cassandra[0m
[0;33mERROR: Migration of keyspace sunbird to version 1.86 failed! Please restore backups and roll back database and code![0m
[0;33mMigration Completed at ==1661348622473[0m
[0;33m[0m
[0;33m[0m
[0;33mSTDERR:[0m
[0;33m[0m
[0;33mSLF4J: Failed to load class "org.slf4j.impl.StaticLoggerBinder".[0m
[0;33mSLF4J: Defaulting to no-operation (NOP) logger implementation[0m
[0;33mSLF4J: See http://www.slf4j.org/codes.html#StaticLoggerBinder for further details.[0m
[0;33mcom.contrastsecurity.cassandra.migration.CassandraMigrationException: Unable to apply migration[0m
[0;33m	at com.contrastsecurity.cassandra.migration.action.Migrate.applyMigration(Migrate.java:147)[0m
[0;33m	at com.contrastsecurity.cassandra.migration.action.Migrate.run(Migrate.java:112)[0m
[0;33m	at com.contrastsecurity.cassandra.migration.CassandraMigration$1.execute(CassandraMigration.java:88)[0m
[0;33m	at com.contrastsecurity.cassandra.migration.CassandraMigration$1.execute(CassandraMigration.java:72)[0m
[0;33m	at com.contrastsecurity.cassandra.migration.CassandraMigration.execute(CassandraMigration.java:201)[0m
[0;33m	at com.contrastsecurity.cassandra.migration.CassandraMigration.migrate(CassandraMigration.java:71)[0m
[0;33m	at com.contrastsecurity.cassandra.migration.utils.MigrationScriptEntryPoint.main(MigrationScriptEntryPoint.java:40)[0m
[0;33mCaused by: com.datastax.driver.core.exceptions.InvalidConfigurationInQueryException: Trigger class 'org.sunbird.cassandra.triggers.TransactionEventTrigger' doesn't exist[0m
[0;33m	at com.datastax.driver.core.exceptions.InvalidConfigurationInQueryException.copy(InvalidConfigurationInQueryException.java:37)[0m
[0;33m	at com.datastax.driver.core.exceptions.InvalidConfigurationInQueryException.copy(InvalidConfigurationInQueryException.java:27)[0m
[0;33m	at com.datastax.driver.core.DriverThrowables.propagateCause(DriverThrowables.java:37)[0m
[0;33m	at com.datastax.driver.core.DefaultResultSetFuture.getUninterruptibly(DefaultResultSetFuture.java:245)[0m
[0;33m	at com.datastax.driver.core.AbstractSession.execute(AbstractSession.java:63)[0m
[0;33m	at com.datastax.driver.core.AbstractSession.execute(AbstractSession.java:39)[0m
[0;33m	at com.contrastsecurity.cassandra.migration.script.CqlScript.execute(CqlScript.java:87)[0m
[0;33m	at com.contrastsecurity.cassandra.migration.resolver.cql.CqlMigrationExecutor.execute(CqlMigrationExecutor.java:35)[0m
[0;33m	at com.contrastsecurity.cassandra.migration.action.Migrate.applyMigration(Migrate.java:145)[0m
[0;33m	... 6 more[0m
[0;33m[0m

TASK [cassandra-deploy : Check the migraion status] ****************************
[0;31mfatal: [10.0.2.102]: FAILED! => {[0m
[0;31m    "changed": false[0m
[0;31m}[0m
[0;31m[0m
[0;31mMSG:[0m
[0;31m[0m
[0;31mFailed migration: ['Migration started at ==1661348554071', 'Cassandra Migration 0.0.1-SNAPSHOT by Contrast Security', 'Connected to cluster: Test Cluster', 'Data center: datacenter1; Host: /10.0.2.102', 'Current version of keyspace sunbird: 1.50', 'Migrating keyspace sunbird to version 1.51 - cassandra', 'Migrating keyspace sunbird to version 1.52 - cassandra', 'Migrating keyspace sunbird to version 1.53 - cassandra', 'Migrating keyspace sunbird to version 1.54 - cassandra', 'Migrating keyspace sunbird to version 1.55 - cassandra', 'Migrating keyspace sunbird to version 1.56 - cassandra', 'Migrating keyspace sunbird to version 1.57 - cassandra', 'Migrating keyspace sunbird to version 1.58 - cassandra', 'Migrating keyspace sunbird to version 1.59 - cassandra', 'Migrating keyspace sunbird to version 1.60 - cassandra', 'Migrating keyspace sunbird to version 1.61 - cassandra', 'Migrating keyspace sunbird to version 1.62 - cassandra', 'Migrating keyspace sunbird to version 1.63 - cassandra', 'Migrating keyspace sunbird to version 1.64 - cassandra', 'Migrating keyspace sunbird to version 1.65 - cassandra', 'Migrating keyspace sunbird to version 1.66 - cassandra', 'Migrating keyspace sunbird to version 1.67 - cassandra', 'Migrating keyspace sunbird to version 1.68 - cassandra', 'Migrating keyspace sunbird to version 1.69 - cassandra', 'Migrating keyspace sunbird to version 1.70 - cassandra', 'Migrating keyspace sunbird to version 1.71 - cassandra', 'Migrating keyspace sunbird to version 1.72 - cassandra', 'Migrating keyspace sunbird to version 1.73 - cassandra', 'Migrating keyspace sunbird to version 1.74 - cassandra', 'Migrating keyspace sunbird to version 1.75 - cassandra', 'Migrating keyspace sunbird to version 1.76 - cassandra', 'Migrating keyspace sunbird to version 1.77 - cassandra', 'Migrating keyspace sunbird to version 1.78 - cassandra', 'Migrating keyspace sunbird to version 1.79 - cassandra', 'Migrating keyspace sunbird to version 1.80 - cassandra', 'Migrating keyspace sunbird to version 1.81 - cassandra', 'Migrating keyspace sunbird to version 1.82 - cassandra', 'Migrating keyspace sunbird to version 1.83 - cassandra', 'Migrating keyspace sunbird to version 1.84 - cassandra', 'Migrating keyspace sunbird to version 1.85 - cassandra', 'Migrating keyspace sunbird to version 1.86 - cassandra', 'ERROR: Migration of keyspace sunbird to version 1.86 failed! Please restore backups and roll back database and code!', 'Migration Completed at ==1661348622473'][0m
[0;31m[0m

PLAY RECAP *********************************************************************
[0;31m10.0.2.102[0m                 : [0;32mok=4   [0m [0;33mchanged=3   [0m unreachable=0    [0;31mfailed=1   [0m [0;36mskipped=1   [0m rescued=0    ignored=0   

[Pipeline] }
[Pipeline] // stage
[Pipeline] }

[Pipeline] // ansiColor
[Pipeline] }
[Pipeline] // stage
[Pipeline] }

[Pipeline] // ansiColor
[Pipeline] ansiColor
[Pipeline] {

[Pipeline] stage
[Pipeline] { (slack_notify)
[Pipeline] sh
+ git rev-parse --short HEAD
[Pipeline] sh
+ git name-rev --name-only HEAD
+ rev
+ cut -d / -f1
+ rev
[Pipeline] sh
+ echo Deploy/dev/Kubernetes/Cassandra
[Pipeline] echo
[33m[1mCould not find env specific Slack channel. Check for global slack channel..[0m
[Pipeline] echo
[33m[1mCould not find slack environment variable. Skipping slack notification..[0m
[Pipeline] }
[Pipeline] // stage
[Pipeline] }

[Pipeline] // ansiColor
[Pipeline] ansiColor
[Pipeline] {

[Pipeline] stage
[Pipeline] { (email_notify)
[Pipeline] sh
+ echo Deploy/dev/Kubernetes/Cassandra
[Pipeline] echo
[33m[1mCould not find env specific email group. Check for global email group..[0m
[Pipeline] echo
[33m[1mCould not find global email group variable. Skipping email notification..[0m
[Pipeline] }
[Pipeline] // stage
[Pipeline] }

[Pipeline] // ansiColor
[Pipeline] }
[Pipeline] // node
[Pipeline] End of Pipeline
ERROR: script returned exit code 2
Finished: FAILURE

**Comment by sharathkashyap** [comment, marked as answer]
@keshavprasadms Need your help here

**Reply by beepdot** [reply]
Please search for similar errors in the forum before posting a new discussion.  Chances are it will be already resolved. Please see this - https://github.com/orgs/Sunbird-Ed/discussions/89
